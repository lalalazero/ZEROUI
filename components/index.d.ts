/// <reference types="react" />
import './index.scss';
declare const message: {
    confirm: (content: string, yes?: (() => void) | undefined, no?: (() => void) | undefined) => void;
    modal: (content: import("react").ReactNode, buttons?: import("react").ReactElement<any, string | ((props: any) => import("react").ReactElement<any, string | any | (new (props: any) => import("react").Component<any, any, any>)> | null) | (new (props: any) => import("react").Component<any, any, any>)>[] | undefined, afterClose?: any) => any;
    alert: (content: string) => void;
};
export { default as Badge } from './badge/index';
export { default as Breadcrumb } from './breadcrumb/index';
export { Button, ButtonGroup } from './button/index';
export { default as Card } from './card';
export { default as Checkbox } from './checkbox/index';
export { default as Dialog } from './dialog/Dialog';
export { default as Col } from './grid/Col';
export { default as Row } from './grid/Row';
export { default as Icon } from './icon/Icon';
export { default as TextInput } from './input';
export { default as NumberInput } from './input-number';
export { default as Content } from './layout/Content';
export { default as Footer } from './layout/Footer';
export { default as Header } from './layout/Header';
export { default as Layout } from './layout/Layout';
export { default as Sider } from './layout/Sider';
export { default as Menu } from './menu/index';
export { default as notification } from './notification/Notification';
export { default as Pagination } from './pagination';
export { default as Radio } from './radio/index';
export { default as Rate } from './rate/index';
export { default as Select } from './select/index';
export { default as Switch } from './switch/index';
export { default as Tooltip } from './tooltip/Tooltip';
export { default as Tree } from './tree/index';
export { message };
