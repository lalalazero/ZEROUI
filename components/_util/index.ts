export function isNumber(number: any): boolean {
    return typeof number === 'number' || (number && number instanceof Number)
}
