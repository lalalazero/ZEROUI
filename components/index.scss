[class^='zeroUI-'] {
    box-sizing: border-box;
    &::before {
        box-sizing: border-box;
    }
    &::after {
        box-sizing: border-box;
    }
}

$padding-left: 20px;
$lighten-border: 1px solid lighten(#ccc, 10%);
$box-shadow-color: #f0f1f2;

// 中性色-浅色背景
$title-text-color-white-bg: lighten(black, 85%);
$primary-text-color-white-bg: lighten(black, 65%);
$second-primary-text-color-white-bg: lighten(black, 45%);
$disabled-color-white-bg: lighten(black, 25%);
$border-color-white-bg: lighten(black, 15%);
$dividers-color-white-bg: lighten(black, 6%);
$background-color-white-bg: lighten(black, 4%);
$table-header-color-white-bg: lighten(black, 2%);

// 中性色-深色背景
$title-text-color-dark-bg: lighten(white, 85%);
$primary-text-color-dark-bg: lighten(white, 65%);
$second-primary-text-color-dark-bg: lighten(white, 45%);
$disabled-color-dark-bg: lighten(white, 30%);
$border-color-dark-bg: lighten(white, 20%);
$dividers-color-dark-bg: lighten(white, 12%);
$background-color-dark-bg: lighten(white, 8%);
$table-header-color-dark-bg: lighten(white, 4%);

// 功能色
$green-6: #52c41a;
$gold-6: #faad14;
$red-5: #ff4d4f;

// cyan 明青色系

$cyan-1: #e6fffb;
$cyan-2: #b5f5ec;
$cyan-3: #87e8de;
$cyan-4: #5cdbd3;
$cyan-5: #36cfc9;
$cyan-6: #13c2c2;
$cyan-7: #08979c;
$cyan-8: #006d75;
$cyan-9: #00474f;
$cyan-10: #002329;

//品牌色
$brand-color: $cyan-6;
$selected-background: $cyan-1;
$hover-color: $cyan-5;
$normal-color: $cyan-6;
$click-color: $cyan-7;

//功能色
$func-link-color: $cyan-6;
$func-success-color: $green-6;
$func-warn-color: $gold-6;
$func-error-color: $red-5;

// 从 zeroview 拷贝过来的全局 scss 变量
$--button-bg: #fff;
$--primary-color: $brand-color;
$--button-active-bg: #eee;
$--border-radius: 4px;
$--color: #333;
$--border-color: #999;
$--border-color-hover: #666;
$active-color: rgb(24, 144, 255);
$active-color-light: rgba(24, 144, 255, 0.1);
$active-bg: #e6f7ff;

$--success-color: #19be6b;
$--warn-color: #f90;
$--error-color: #ed4014;
$--info-color: #2db7f5;

$height: 32px;
$border-color: #999;
$border-color-hover: #666;
$border-color-1: #ccc;
$border-color-ant-design: #ebedf0;
$border-radius: 4px;
$font-size: 14px;
$font-size-lg: 16px;
$font-size-sm: 12px;
$box-shadow-color: rgba(10, 7, 7, 0.5);
$error-color: #f1453d;

$bg-white: #fff;
$bg-black: #000;
$bg-gray-light: #ccc;

$disabled-color: lighten(gray, 20%);

$toast-bg: rgba(0, 0, 0, 0.75);
$toast-height: 40px;
$animation-duration: 300ms;

$table-border-color: rgb(232, 232, 232);
$table-strip-color: #fafafa;
$table-row-hover-color: rgb(230, 247, 255);

%box-shadow {
    box-shadow: 0 0 6px 0 #ccc;
}
%pos-center {
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}
%text-ellipsis {
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
}

@keyframes spin {
    0% {
        transform: rotate(0deg);
    }
    100% {
        transform: rotate(360deg);
    }
}

$spinAnimation: spin 1s infinite linear;

// 抄 antd 的变量设计

$font-size-base: 14px;
$font-size-lg: $font-size-base + 2px;
$font-size-sm: 12px;

// vertical paddings
$padding-lg: 24px; // containers
$padding-md: 16px; // small containers and buttons
$padding-sm: 12px; // Form controls and items
$padding-xs: 8px; // small items

$btn-height-base: 32px;
$btn-height-lg: 40px;
$btn-height-sm: 24px;

$btn-padding-base: 0 $padding-md - 1px;
$btn-font-size-lg: $font-size-lg;
$btn-font-size-sm: $font-size-base;
$btn-padding-lg: $btn-padding-base;
$btn-padding-sm: 0 $padding-xs - 1px;

$btn-disabled-color: #f5f5f5;
