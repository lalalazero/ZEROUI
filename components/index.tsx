import { Button, ButtonGroup } from './button/index'
import Dialog, { alert, confirm, modal } from './dialog/Dialog'
import Icon from './icon/Icon'
import './index.scss'
import Content from './layout/Content'
import Footer from './layout/Footer'
import Header from './layout/Header'
import Layout from './layout/Layout'
import Sider from './layout/Sider'
import { notification } from './notification/Notification'

const message = {
    confirm,
    modal,
    alert,
}

export { default as Col } from './grid/Col'
export { default as Row } from './grid/Row'
export { default as Input } from './input/index'
export { default as Menu } from './menu/index'
export { default as Notification } from './notification/Notification'
export { default as Pagination } from './pagination'
export { default as Select } from './select/index'
export { default as Tooltip } from './tooltip/Tooltip'
export {
    Icon,
    Dialog,
    Layout,
    Content,
    Header,
    Footer,
    Sider,
    message,
    Button,
    ButtonGroup,
    notification,
}
