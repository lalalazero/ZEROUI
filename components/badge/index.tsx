import React, { CSSProperties, useMemo } from 'react'
import { isNumber } from '../_util'
import { classname } from '../_util/classes'
import './style.scss'

const PREFIX = 'zeroUI-badge'

export interface BadgeProps extends CSSProperties {
    count?: number
    className?: string
    showZero?: boolean
    overCount?: number
}

const Badge: React.FC<BadgeProps> = (props) => {
    const { className, children, showZero, overCount, count, ...style } = props

    const isSolo = !children

    const isCustom = !isNumber(count)

    const isOvered =
        isNumber(count) &&
        isNumber(overCount) &&
        (count as number) > (overCount as number)

    const getCount = useMemo(() => {
        if (isNumber(count)) {
            const countNum = count as number
            const max = overCount as number

            if (countNum < max) {
                return countNum
            }

            if (countNum > max) {
                return max + '+'
            }
        }

        return count
    }, [props.showZero, props.count, props.overCount])

    const shouldRenderIcon = count !== 0 || (count === 0 && showZero)

    return (
        <div className={classname(PREFIX + '-wrapper')}>
            {isSolo ? (
                <span
                    className={classname(PREFIX + '-solo', className)}
                    {...style}
                >
                    {getCount}
                </span>
            ) : (
                <>
                    {shouldRenderIcon && (
                        <span
                            className={
                                isCustom
                                    ? classname(PREFIX + '-custom', {
                                          [`${PREFIX}-over-max`]: isOvered,
                                      })
                                    : classname(PREFIX, {
                                          [`${PREFIX}-over-max`]: isOvered,
                                      })
                            }
                        >
                            {getCount}
                        </span>
                    )}

                    {children}
                </>
            )}
        </div>
    )
}

Badge.defaultProps = {
    showZero: false,
    overCount: 99,
}
export default Badge
