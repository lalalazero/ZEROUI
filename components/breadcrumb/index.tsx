import React from 'react'
import { classname } from '../_util/classes'
import './style.scss'

const PREFIX = 'zeroUI-breadcrumb'

const Item: React.FC<any> = (props) => {
    return (
        <div className={classname(PREFIX)}>
            {props.children}
            <span className={classname(`${PREFIX}-seperator`)}>/</span>
        </div>
    )
}

export interface BreadcrumbInterface extends React.FC<any> {
    Item: typeof Item
}
const Breadcrumb: BreadcrumbInterface = (props) => {
    return (
        <div className={classname(`${PREFIX}-wrapper`)}>{props.children}</div>
    )
}

Breadcrumb.Item = Item

export default Breadcrumb
