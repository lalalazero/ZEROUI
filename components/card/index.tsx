import React, { HTMLAttributes, ReactNode } from 'react'
import { classname } from '../_util/classes'
import './style.scss'

const cardPrefix = 'zeroUI-card'
const gridPrefix = 'zeroUI-card-grid'

export interface CardProps
    extends Omit<HTMLAttributes<HTMLDivElement>, 'title' | 'size'> {
    title?: ReactNode
    size?: 'small'
    bordered?: boolean
    hoverable?: boolean
}

export interface CardGridProps {
    hoverable?: boolean
    style?: React.CSSProperties
}

const Grid: React.FC<CardGridProps> = (props) => {
    const { hoverable = true, children, style } = props

    const classes = classname(gridPrefix, {
        [`${gridPrefix}-hoverable`]: hoverable,
    })
    return (
        <div className={classes} style={style}>
            {children}
        </div>
    )
}

export interface CardInterface extends React.FC<CardProps> {
    Grid: typeof Grid
}

const Card: CardInterface = (props) => {
    const {
        title,
        children,
        size = '',
        bordered = true,
        hoverable = false,
        ...rest
    } = props

    const isContainGrid = () => {
        let containGrid = false
        React.Children.forEach(children, (element: JSX.Element) => {
            if (element && element.type && element.type === Grid) {
                containGrid = true
            }
        })

        return containGrid
    }

    const classes = classname(cardPrefix, `${cardPrefix}-${size}`, {
        [`${cardPrefix}-noborder`]: !bordered,
        [`${cardPrefix}-hoverable`]: hoverable,
        [`${cardPrefix}-has-grid`]: isContainGrid(),
    })

    return (
        <div className={classes} {...rest}>
            {title && (
                <div className={classname(cardPrefix + '-title-panel')}>
                    {title}
                </div>
            )}

            <div className={classname(cardPrefix + '-content-panel')}>
                {children}
            </div>
        </div>
    )
}

Card.Grid = Grid

export default Card
