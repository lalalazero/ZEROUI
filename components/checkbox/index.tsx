import React, { useEffect, useRef, useState } from 'react'
import { classname } from '../_util/classes'
import CheckboxGroup from './CheckboxGroup'
import './style.scss'

const prefix = 'zeroUI-checkbox'

export interface CheckboxInterface extends React.FC<CheckBoxProps> {
    Group: typeof CheckboxGroup
}

interface CheckBoxProps {
    name?: string
    checked?: boolean
    onChange?: (name: string, checked: boolean) => void
    disabled?: boolean
    indeterminate?: boolean
}
const Checkbox: CheckboxInterface = (props = {}) => {
    const ref = useRef<HTMLInputElement>(null)
    const { disabled = false } = props
    const [checked, setChecked] = useState(false)

    useEffect(() => {
        if (typeof props.checked === 'boolean') {
            setChecked(props.checked)
            if (ref.current) {
                ref.current.checked = props.checked
            }
        }
    })

    useEffect(() => {
        if (typeof props.indeterminate === 'boolean' && ref.current) {
            ref.current.indeterminate = props.indeterminate
        }
    }, [props.indeterminate])

    const handleChange: React.ChangeEventHandler<HTMLInputElement> = (
        event
    ) => {
        const newCheck = event.target.checked
        setChecked(newCheck)
        props.onChange && props.onChange(props.name || '', newCheck)
    }

    return (
        <span
            className={classname(prefix + '-wrapper')}
            data-disabled={disabled.valueOf()}
        >
            <input
                type="checkbox"
                ref={ref}
                className={classname(prefix)}
                name={props.name}
                checked={checked}
                onChange={handleChange}
                disabled={disabled}
            ></input>
            <label
                htmlFor={props.name}
                className={classname(prefix + '-label')}
            >
                {props.children}
            </label>
        </span>
    )
}

Checkbox.Group = CheckboxGroup

export default Checkbox
