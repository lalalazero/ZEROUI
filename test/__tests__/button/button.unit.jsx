import React from 'react'
import renderer from 'react-test-renderer';
import { mount } from 'enzyme'
import { Button, ButtonGroup } from '../../../components/index'
import mountTest from '../../mountTest'

describe('button 测试', () => {
    mountTest(Button)
    mountTest(() => <Button size="large"></Button>)
    mountTest(() => <Button size="small"></Button>)
    mountTest(ButtonGroup)
    it('renders ok', () => {
        const button = renderer.create(<Button>button</Button>)
        let tree = button.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('可以设置icon', () => {
        const wrapper = renderer.create(<Button icon="wechat">button</Button>)
        let tree = wrapper.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('可以设置loading', () => {
        const wrapper = renderer.create(<Button icon="wechat" loading={true}></Button>)
        let tree = wrapper.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('图标可以设置在右侧', () => {
        const wrapper = mount(<Button icon="wechat" position='right'>button</Button>)
        expect(wrapper.find('.zeroUI-button-right')).toHaveLength(1)
    })

    it('正常状态可以click', () => {
        const fn = jest.fn()
        const wrapper = mount(<Button onClick={fn}>Button</Button>)
        wrapper.simulate('click')
        expect(fn).toHaveBeenCalled()
    })

    it('disabled时不可以click', () => {
        const fn = jest.fn()
        const wrapper = mount(<Button onClick={fn} disabled>Button</Button>)
        wrapper.simulate('click')
        expect(fn).not.toHaveBeenCalled()
    })
})