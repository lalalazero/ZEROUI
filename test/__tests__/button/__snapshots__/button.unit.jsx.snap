// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button 测试 renders ok 1`] = `
<button
  className="zeroUI-button zeroUI-button-normal zeroUI-button-left"
>
  
  
  <div
    className="zeroUI-button-content"
  >
    button
  </div>
</button>
`;

exports[`button 测试 可以设置icon 1`] = `
<button
  className="zeroUI-button zeroUI-button-normal zeroUI-button-left"
>
  
  <svg
    className="zeroUI-icon"
  >
    <use
      xlinkHref="#icon-wechat"
    />
  </svg>
  <div
    className="zeroUI-button-content"
  >
    button
  </div>
</button>
`;

exports[`button 测试 可以设置loading 1`] = `
<button
  className="zeroUI-button zeroUI-button-normal zeroUI-button-left zeroUI-button-loading"
>
  <svg
    className="zeroUI-icon"
  >
    <use
      xlinkHref="#icon-loading"
    />
  </svg>
  
  <div
    className="zeroUI-button-content"
  />
</button>
`;
